### ISSUE TITLE
[Entrega 1] Fallo en test case 

### ISSUE THREAD

**Issue by stgo32** [RodrigoToroIcarte/IIC2113-2023-1#7]
Dos cosas:

1. El test 01 de los inválidos sale como correcto cuando en realidad está incorrecto debido a que tiene 4 cartas "Irish Wip".
2. Rider no me deja correr los test cases automáticos, la casilla sale desactivada como se muestra en la foto, ¿saben como arreglarlo?

![image](https://user-images.githubusercontent.com/69644287/228032024-b6cfd904-8f63-40ba-b9c9-d6580a298134.png)

Gracias

**Comment by RodrigoToroIcarte** [comment]
¡Hola!

Acá van las respuestas a tus preguntas:

1. El test case está correcto. La carta "irish whip" es un set-up por lo que puedes tener cuantas copias quieras de esa carta en tu mazo.
2. A veces a rider le cuesta cachar que un proyecto es de tipo test. Una posible solución es que abras el archivo de test y corras uno de los test directo:
![Screenshot from 2023-03-27 16-31-55](https://user-images.githubusercontent.com/7000518/228047185-f333f1ac-d075-4923-b8d2-f6639cf67ca3.png)

**Comment by stgo32** [comment]
Gracias! ahora los test si corren

**Comment by sawiiii** [comment]
Hola!,

El test-01 de los inválidos es correcto?

**Comment by RodrigoToroIcarte** [comment]
Sí, todos los test cases están correctos. El test case 1 parte cargando el mazo `data/02-InvalidDecks/01.txt`. A simple vista, ese mazo pareciera ser inválido porque tiene cuatro cartas `Irish Whip`. Eso rompe la regla de tener un máximo de 3 copias de la misma carta. Pero como dice el enunciado, esta regla tiene 2 excepciones:

![Screenshot from 2023-03-28 12-07-34](https://user-images.githubusercontent.com/7000518/228282936-e3b1da72-5968-44d0-8f6c-a3428da5ca0f.png)

La carta `Irish Whip` tiene el subtipo `SetUp`. Es por esto que el mazo es válido a pesar de tener 4 `Irish Whip`.

**Comment by sawiiii** [comment]
Eso es lo que me causa duda, no debería estar el mazo entonces en los test válidos? (Lo tengo bien, pero causa duda).

**Comment by RodrigoToroIcarte** [comment]
Aaaah, ahora entiendo la duda. Lo que pasa es que me vi en la necesidado de poner un mazo válido entre los mazos inválidos porque de lo contrario era muy fácil pasar todos los tests de los casos inválidos. Bastaba con siempre decir que el mazo era inválido usando el siguiente código:

```
_view.AskUserToSelectDeck(_deckFolder);
_view.SayThatDeckIsInvalid();
```

... y con solo eso llegan al 3.0 según la rúbrica :P. Así que metí un mazo válido para que a veces el primer mazo entregado fuese válido, pero el segundo fuese inválido. Es probable que existan mejores formas de resolver ese problema, pero estaba cansado y lo hice así no más :)

**Comment by sawiiii** [comment]
Profe, consulta es posible que el método AskUserToSelectDeck(), duplique la cantidad de archivos?, porque la hacerlo me da este output:
<img width="338" alt="Screenshot 2023-03-28 at 12 54 27" src="https://user-images.githubusercontent.com/69648324/228296368-572114da-38c0-47c9-9311-2da39bc8478a.png">

Estoy casi seguro de no haber tocado el código🥲

**Comment by RodrigoToroIcarte** [comment]
Hoy hablé (y solucioné) justo ese problema con un estudiante al final de la clase. Eres tú ese estudiante o eres otra persona con el mismo problema??

**Comment by sawiiii** [comment]
Soy el mismo profe!
